src/shogun/io/ARFFFile.cpp

```diff
@@ -0,0 +1,357 @@
+			m_attribute_names.emplace_back(name);
+		}
+		// comments in this section are ignored
+		else if (m_current_line.substr(0, 1) == m_comment_string)
+		{
+		}
+		else if (
+		    to_lower(m_current_line.substr(0, strlen(m_data_string))) ==
+		    m_data_string)
+			m_state = true;
+	};
+
+	auto check_attributes = [this]() {
+		// attributes cannot be empty
+		return !m_attributes.empty();
+	};
+	process_chunk(read_attributes, check_attributes, true);
+
+	auto read_data = [this]() {
+		// it's a comment and can be skipped
+		if (SG_UNLIKELY(m_current_line.substr(0, 1) == m_comment_string))
+			return;
+		// it's the data string (i.e. @data"), does not provide information
+		if (SG_UNLIKELY(
+		        to_lower(m_current_line.substr(0, strlen(m_data_string))) ==
+		        m_data_string))
+		{
+			return;
+		}
+		// assumes that until EOF we should expect comma delimited values
+		std::vector<std::string> elems;
+		split(m_current_line, ",", std::back_inserter(elems), "\'\"");
+		// only parse rows that do not contain missing values
+		if (std::find(elems.begin(), elems.end(), "?") == elems.end())
+		{
+			auto nominal_pos = m_nominal_attributes.begin();
+			auto date_pos = m_date_formats.begin();
+			int i = 0;
+			for (; i < elems.size(); ++i)
+			{
+				Attribute type = m_attributes[i];
+				switch (type)
+				{
+				case (Attribute::Numeric):
+				case (Attribute::Integer):
+				case (Attribute::Real):
+				{
+					try
+					{
+						shogun::get<std::vector<float64_t>>(m_data_vectors[i])
+						    .push_back(std::stod(elems[i]));
+					}
+					catch (const std::invalid_argument&)
+					{
+						SG_SERROR(
+						    "Failed to covert \"%s\" to numeric on line %d.\n",
+						    elems[i].c_str(), m_line_number)
+					}
+				}
+				break;
+				case (Attribute::Nominal):
+				{
+					if (nominal_pos == m_nominal_attributes.end())
+						SG_SERROR(
+						    "Unexpected nominal value \"%s\" on line %d\n",
+						    elems[i].c_str(), m_line_number);
+					auto encoding = (*nominal_pos).second;
+					auto trimmed_el = trim(elems[i]);
+					remove_char_inplace(trimmed_el, '\'');
+					auto pos =
+					    std::find(encoding.begin(), encoding.end(), trimmed_el);
+					if (pos == encoding.end())
+						SG_SERROR(
+						    "Unexpected value \"%s\" on line %d\n",
+						    trimmed_el.c_str(), m_line_number);
+					float64_t idx = std::distance(encoding.begin(), pos);
+					shogun::get<std::vector<float64_t>>(m_data_vectors[i])
+					    .push_back(idx);
+					nominal_pos = std::next(nominal_pos);
+				}
+				break;
+				case (Attribute::Date):
+				{
+					date::sys_seconds t;
+					std::istringstream ss(elems[i]);
+					if (date_pos == m_date_formats.end())
+						SG_SERROR(
+						    "Unexpected date value \"%s\" on line %d.\n",
+						    elems[i].c_str(), m_line_number);
+					ss >> date::parse(*date_pos, t);
+					if (bool(ss))
+					{
+						auto value_timestamp = t.time_since_epoch().count();
```

Comment:
I wonder whether this is a good thing to do ...

Reply:
Until we have meta-data support, we can do it like this. But maybe we can offer a utility function (swig-ed) that can convert from this format to a date string and vice versa?

Reply:
yup, that's a good idea! You mean a class function that has something like `get_time_as_string()`?

Reply:
yes something like that...also maybe a vectorized one that converts a whole vector/matrix

Reply:
(because the arff parser generates exactly that)